src/Executor.cc

```diff
@@ -72,12 +72,12 @@ void Executor::cancel()
         return;
 
     comm_channel_->immediate_termination_requested_ = true;
-    context_ = future_.get(); // Wait for thread to join
+    context_.variables = future_.get(); // Wait for thread to join
     comm_channel_->immediate_termination_requested_ = false;
 }
 
-Context Executor::execute_sequence(Sequence sequence, Context context,
-                                std::shared_ptr<CommChannel> comm) noexcept
+VariableTable Executor::execute_sequence(Sequence sequence, Context context,
+                                         std::shared_ptr<CommChannel> comm) noexcept
```

Comment:
I think I do not get the style used here. Lining up behind the `(`?

Otoh in `Executor.h` line 198/200 you remove all the blanks and indent it just one 'tab' (i.e. 4 blanks).

Reply:
Occasionally I like to line up parameters behind the `(`. IMO it makes the declaration much more readable. I guess it would not survive clang-format, though. :)

Reply:
Just to note ... I _tend_ to write this as follows

```
___ ___ funktion_name(argument, argument,
___ ___ ___ ___ argument, ...) {
___ ___ ___ code();
___ ___ }
```
or

```
___ ___ funktion_name(argument, argument,
___ ___ ___ ___ argument, ...)
___ ___ {
___ ___ ___ code();
___ ___ }
```

Vertical alignment means always you might need to touch unchanged lines in commits, just to get the alignment repaired.

Checking `Executor.cc` that really seems to be your style. Hmm. /shrug